tooling/templatize/pkg/ev2/mapping.go

```diff
@@ -7,23 +7,65 @@ import (
 	"github.com/Azure/ARO-HCP/tooling/templatize/pkg/config"
 )
 
-func EV2Mapping(input config.Variables, prefix []string) (map[string]string, map[string]interface{}) {
+type PlaceholderGenerator func(key []string, value interface{}) (flattenedKey string, replaceVar string)
+
+// NewDunderPlaceholders returns a PlaceholderGenerator function that generates
+// placeholder strings by joining the provided key elements with underscores
+// and surrounding them with double underscores.
+//
+// The returned function takes a key (a slice of strings) and an unused interface{} value,
+// and returns a flattenedKey and replaceVar, both of which are the generated placeholder string.
+//
+// Example:
+//
+//	key := []string{"foo", "bar"}
+//	flattenedKey, replaceVar := NewDunderPlaceholders()(key, nil)
+//	// flattenedKey and replaceVar will both be "__foo_bar__"
+func NewDunderPlaceholders() PlaceholderGenerator {
+	return func(key []string, _ interface{}) (flattenedKey string, replaceVar string) {
+		flattenedKey = fmt.Sprintf("__%s__", strings.Join(key, "_"))
```

Comment:
can we change separator from "_" to "."?

Reply:
done. this has the benefit of making the generated placeholders more unique. while we have the occasional `_` in a var name, we don't have dots there.